src/bosh-monitor/lib/bosh/monitor/plugins/resurrector.rb

```diff
@@ -98,6 +103,27 @@ def process(alert)
 
       private
 
+      def scan_and_fix_already_queued_or_processing?(deployment_name)
+        url = @uri.dup
+        url.path = '/tasks'
+        if auth_provider(director_info).auth_header.is_a?(Array)
+          auth_header_value = "Basic #{Base64.strict_encode64("#{auth_provider(director_info).auth_header[0]}:#{auth_provider(director_info).auth_header[1]}")}"
+        else
+          auth_header_value = auth_provider(director_info).auth_header
+        end
+        headers = {
+          'Authorization' => auth_header_value,
+          'Content-Type' => 'application/json',
+        }
+        url.query = URI.encode_www_form({ deployment: deployment_name, state: 'queued,processing', verbose: 2 })
+        response = send_http_get_request(url.to_s, headers)
+        return true unless JSON.parse(response.body).select do |item|
+                             item['description'] == 'scan and fix'
+                           end.empty? && response.status_code == 200
```

Comment:
This statement here is a little bit hard to understand but I'm not sure whether we should return `true` when the `status_code` isn't 200. This is the case at the moment here. Additionally, we don't have a test case which specify what should be the response in case of `status_code` different than 200.

Reply:
Thanks. Yeah you're right, it is hard to read. 

The idea was to not add an additional task if we currently cannot query the director for tasks. 

The scenario for this PR is one where the director is potentially dealing with 100s of missing vms. So the general load may be high and the tasks endpoint maybe affected. Scheduling additional tasks in that scenario would worsen the situation. The above code would eventually schedule a scan and fix when it was able to check current tasks ( status_code == 200) and it's sure there is no `scan and fix` ( empty? == true ).

I could rewrite it to:

```+        tasks_response = send_http_get_request(url.to_s, headers)
+
+        logger.debug(tasks_response.body)
+        queued_scan_and_fix = JSON.parse(tasks_response.body).select do | task |
+          task['description'] == 'scan and fix'
+        end
+
+        return false if tasks_response.status_code == 200 && queued_scan_and_fix.empty?```

which probably makes it easier to understand but I'm not sure we should schedule a task if the status code isn't 200


in the case where the response isn't 200, the resurrector would skip sending a `scan and fix` for this iteration but still would do it in the next iteration (given that we can properly acccess the tasks endpoint).

Reply:
Ok, this makes sense. It will be great if you can implement the suggestion you provided and also add a comment why we don't want to schedule new tasks when status codes different than 200 or a debug log that we are skipping the scheduling because of that.

Reply:
Done.